draft-irtf-cfrg-opaque.md

```diff
@@ -384,26 +390,49 @@ OPAQUE relies on the following protocols and primitives:
 
-# Core Protocol {#protocol}
+# Offline Registration {#offline-phase}
 
-OPAQUE consists of two stages: registration and authenticated key exchange.
-In the first stage, a client registers its password with the server and stores
-its encrypted credentials on the server. In the second stage, a client obtains
-those credentials, unlocks them using the user's password and subsequently uses
-them as input to an AKE protocol.
+Registration is executed between a user U (running on a client machine) and a
+server S. It is assumed S can identify U and the client can
+authenticate S during this registration phase. This is the only part
+in OPAQUE that requires an authenticated channel, either physical, out-of-band,
+PKI-based, etc. This section describes the registration flow, message encoding,
+and helper functions. Moreover, U has a key pair (skU, pkU) for an AKE protocol
+which is suitable for use with OPAQUE; See {{online-phase}}. (skU, pkU) may be
+randomly generated for the account or provided by the calling client.
+Clients MUST NOT use the same key pair (skU, pkU) for two different accounts.
 
-Both registration and authenticated key exchange stages require running an OPRF protocol.
-The latter stage additionally requires running a mutually authenticated
-key exchange protocol using credentials recovered after the OPRF protocol completes.
-(The key exchange protocol MUST satisfy forward secrecy and the KCI requirement
-discussed in {{security-considerations}}.)
+To begin, U chooses password pwdU, and S chooses its own pair of private-public
+keys skS and pkS for use with the AKE. S can use the same pair of keys with
+multiple users. These steps can happen offline, i.e., before the registration phase.
+Once complete, the registration process proceeds as follows:
+
+~~~
+ Client (pwdU, creds)                               Server (skS, pkS)
+ --------------------------------------------------------------------
+ (request, blind) = CreateRegistrationRequest(pwdU)
+
+                               request
+                      ------------------------->
+
+            (response, kU) = CreateRegistrationResponse(request, pkS)
 
-We first define the core OPAQUE protocol based on a generic OPRF, hash, and MHF function.
-{{instantiations}} describes specific instantiations of OPAQUE using various AKE protocols,
-including: HMQV, 3DH, and SIGMA-I. {{I-D.sullivan-tls-opaque}} discusses integration with
-TLS 1.3 {{RFC8446}}.
+                               response
+                      <-------------------------
+
+ (record, export_key) = FinalizeRequest(pwdU, creds, blind, response)
 
-## Credential types and envelope construction {#data-types}
+                                record
+                      ------------------------->
+~~~
+
+Both client and server MUST validate the other party's public key before use.
+See {{validation}} for more details.
+
+Upon completion, the S stores U's credentials for later use. See {{credential-file}}
```

Comment:
Nit: "the S stores U's credentials" -> "S stores U's credentials" / "the server stores the user's credentials"